feat: allow passing Netlify API host to Netlify API client

Benaiah opened the pull request:
**Which problem is this pull request solving?**

We currently can't run netlify-build against the staging version of our API for testing certain workflows. This solves that by adding an `--api-host` command line flag that can be used to specify the hostname of the API.

**List other issues or pull requests related to this problem**

https://github.com/netlify/buildbot/issues/1172

**Checklist**

Please add a `x` inside each checkbox:

- [x] I have read the [contribution guidelines](../blob/master/CONTRIBUTING.md).
- [x] The status checks are successful (continuous integration). Those can be seen below.

Benaiah commented:
@ehmicky this is ready for you to take a look when you get a chance!

ehmicky commented on packages/build/src/core/constants.js:
```diff
@@ -73,6 +74,10 @@ const getConstants = async function ({
+    /**
+     * The Netlify API endpoint
+     */
+    NETLIFY_API_HOST: apiHost,
```
This will be exposed to plugin authors, is this intentional?

Benaiah replied:
This is intentional because this would allow properly developed plugins to be tested in staging. It's not vital by any means, tho.

ehmicky replied:
Thanks for the explanation @Benaiah.
Should this be documented or hidden?

Benaiah replied:
I think this should be documented, since we'd ideally like plugin authors to respect it.

ehmicky replied:
I agree :+1: 
Should we create an issue for it in `netlify/docs`?

cc @rstavchansky @verythorough

rstavchansky replied:
I'm interested to hear from @verythorough on whether we'd expect external Build Plugin developers to test against a staging server. If so, @Benaiah I'd appreciate it very much if you could create a docs issue to track this update.

Benaiah replied:
My thought was that we'd want to be able to test build plugins on staging for internal use, not that external authors would be using our staging system.

Benaiah replied:
We've decided to not expose this env variable for now and continue the discussion in https://github.com/netlify/build/issues/2373.

ehmicky reviewed (commented):
This is looking great :tada: 

The test seems to be failing on Node 8, do you know why that is?

Benaiah commented:
This is showing up in the diff of the failed snapshot:

```
   -   Failed retrieving user account: Unauthorized. Double-check your login status with 'netlify status' or contact support with details of your error
```

It looks like a snapshot where an unexpected error is occurring has been inadvertently stored in the PR.

Benaiah commented:
I've rebased it again and it looks like the error is still happening. It almost seems like the test server is forwarding the request to the actual API and getting an error from it, but I'm not sure why that's happening when my `startServer` use looks the same as in other tests and I'm also not sure why it would only be happening when they're running in node 8. It is the test I added that's causing the problem, but I'm kind of stumped as to why.

verythorough commented:
I just switched labels and realized that the reason the bot thought this was a feature was because of the PR name. Since this is a change that improves our internal development, isn't it a chore?

ehmicky commented:
@verythorough Does this depend on whether `NETLIFY_API_HOST` should be exposed to plugin authors? 
What are your thoughts about exposing it or not?

ehmicky commented:
PR solving the reason why tests are failing on Node 8 in CI on this PR: https://github.com/netlify/build/pull/2364

Benaiah commented:
Tests are passing now - thanks @ehmicky for figuring that out!

Benaiah commented:
@ehmicky can I get a thumbs up on this now that we've addressed the env variable issue?

ehmicky reviewed (approved):
Yes!

erezrokah commented:
I think this might have broken the way the CLI passes the api host to `@netlify/config`.
I had to make this change https://github.com/netlify/cli/pull/1836/commits/e28dedd6d2abe4ba211277e9c864062d89da8ca7#diff-4277e3cb4bc6c06fc1915215a34bb00ec936a04102d58ea72b4705fc988ad083R150 to get the tests on my PR use the mock API again.

erezrokah commented:
I think we can silently fix it in the CLI no worries. Let me open a PR for it

ehmicky commented:
I opened a PR to rename `apiHost` to `host` because this renaming technically breaks semver: https://github.com/netlify/build/pull/2379
